packages/plugin-workspace-tools/sources/commands/workspace.ts

```diff
@@ -0,0 +1,79 @@
+import {WorkspaceRequiredError}                            from "@berry/cli";
+import {CommandContext, Configuration, Project, Workspace} from "@berry/core";
+import {structUtils}                                       from "@berry/core";
+import {Command, UsageError}                               from "clipanion";
+
+// eslint-disable-next-line arca/no-default-export
+export default class WorkspaceCommand extends Command<CommandContext> {
+  @Command.String()
+  workspaceName!: string;
+
+  @Command.String()
+  commandName!: string;
+
+  @Command.Proxy()
+  args: Array<string> = [];
+
+  static usage = Command.Usage({
+    category: `Workspace-related commands`,
+    description: `run a command on the specified namespace`,
+    details: `
+      This command will run a given sub-command on a single workspace.
+    `,
+    examples: [
+      [
+        `Add a package to a single workspace`,
+        `yarn workspace components add -D react`,
+      ],
+      [
+        `Run build script on a single workspace`,
+        `yarn workspace components run build`,
+      ],
+    ],
+  });
+
+  @Command.Path(`workspace`)
+  async execute() {
+    const configuration = await Configuration.find(
+      this.context.cwd,
+      this.context.plugins
+    );
+    const {project, workspace: cwdWorkspace} = await Project.find(
+      configuration,
+      this.context.cwd
+    );
+
+    if (!cwdWorkspace) throw new WorkspaceRequiredError(this.context.cwd);
+
+    const candidates = project.workspaces;
+    const candidatesByName = new Map(
+      candidates.map(
+        (workspace): [string, Workspace] => {
+          const ident = structUtils.convertToIdent(workspace.locator);
+          return [structUtils.stringifyIdent(ident), workspace];
+        }
+      )
+    );
+
+    const workspace = candidatesByName.get(this.workspaceName);
+
+    if (workspace === undefined) {
+      const otherNames = Array.from(candidatesByName.keys()).sort();
+      throw new UsageError(
+        `Workspace '${
+          this.workspaceName
+        }' not found. Did you mean any of the following:\n  - ${otherNames.join(
+          "\n  - "
+        )}?`
+      );
+    }
+
+    return this.cli.run([this.commandName, ...this.args], {
+      cwd: workspace.cwd,
+      plugins: this.context.plugins,
+      stderr: this.context.stderr,
+      stdin: this.context.stdin,
+      stdout: this.context.stdout,
```

Comment:
I think you only need to pass `cwd`, the rest will be implicitly forwarded.

Reply:
Just saw it in the autocomplete. Looked at other callsites and they don't forward the context either. Should be good.

Reply:
Done